modules/sc-mesh-secure-deployment/src/nats/src/cbma_controller.py

```diff
@@ -337,15 +341,15 @@ def __shutdown_and_delete_bridge(self, bridge_name):
             ip.close()
 
     def __cleanup_cbma(self):
-        self.__shutdown_interface(self.lower_batman)
-        self.__shutdown_interface(self.upper_batman)
+        self.__shutdown_interface(self.LOWER_BATMAN)
+        self.__shutdown_interface(self.UPPER_BATMAN)
         self.__delete_ebtables_rules()
         self.__delete_macsec_links()
-        self.__destroy_batman_interface(self.lower_batman)
-        self.__destroy_batman_interface(self.upper_batman)
+        self.__destroy_batman_interface(self.LOWER_BATMAN)
+        self.__destroy_batman_interface(self.UPPER_BATMAN)
         # HACK: How can we know that "br-upper" even exists and who owns it?
```

Comment:
It shouldn't: IIRC it was the name of the MACsec bridge in the Upper CBMA, but as part of the rewrite this is being renamed to the conflict-avoiding Xmb002091abcdef format where X is 'l' in Lower and 'u' in Upper, and 00:20:91:ab:cd:ef is the MAC address of the underlying interface. This is also for coherence with the other interfaces created as part of the "mess" between black & white or between white & red: MACVLAN interfaces prefixed with Xmv and MACsec interfaces prefixed with Xms (except Xmv and Xms names contain the _peer's_ MAC, so on a node called A under the umb002091aaaaaa bridge for example you might see ums002091bbbbbb and umv002091bbbbbb for a B peer, um[sv]002091cccccc for a C peer, and so on). The mess of [lu]m[bsv]* interfaces is handled within CBMA, CommsController only tells CBMA something like "you're in charge of between wlan0 and bat0", another "between bat0 and bat1", yet another "between halow0 and bat1" and so on.

TL;DR: please delete br-upper :)

Reply:
Deletion of br-upper has to be kept for now in cbma_controller's __cleanup_cbma method as current version of CBMA creates it and in case we restart MDM agent we need to ensure that everything CBMA related is cleared before trying to set it up again. I've mentioned it to be a hack as CBMA feature code is the one who has possibly created it (same applies to ebtables). Logically CBMA feature code should take care of cleaning what is has created when CBMA processes are terminated. Once CBMA rewrite code available for integration we can remove that.